.github/workflows/test.yml

```diff
@@ -113,15 +99,33 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
-      - uses: actions/download-artifact@v2
-        with:
-          name: parallel-lint-phar
-
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          coverage: false
+          coverage: none
+          tools: cs2pr
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v1
+
+      - name: 'Run unit tests PHP 5.4, 5.5'
+        if: ${{ matrix.php < 5.6 }}
+        run: composer testphp5
+
+      - name: 'Run unit tests PHP >= 5.6'
+        if: ${{ matrix.php >= 5.6 }}
+        run: composer test
+
+      - name: 'Integration test 1 - linting own code, no colors'
+        run: ./parallel-lint --exclude vendor --exclude tests/examples --no-colors .
+
+      - name: 'Integration test 2 - linting own code'
+        run: ./parallel-lint --exclude vendor --exclude tests/examples . --checkstyle | cs2pr
```

Comment:
if the lint own code fails, you won't get cs2pr results to pull the request as the erroring job cancels other jobs. you probably want to move cs2pr filtered job up.

And due to `| ` filtering, the second command won't even trigger the error code to shell.

```sh
➔ sh -c 'false; echo $?'
1
➔ sh -c 'false | cat; echo $?'
0
```

note, cs2pr itself may trigger error if it finds error in input, haven't checked that.

so if you fix the workflow based on that, be sure include the reasonning in the .yaml as a comment, very unlikely somebody will find this comment box when refactoring that part in there.
maybe just link to this post.

Reply:
`cs2pr` has an option that insinuates the script exits if it finds any errors or warnings, so we should be good with the pipe.

> `--graceful-warnings`: Don't exit with error codes if there are only warnings

Reply:
yeah, but please read the first part of my message: but no reports to github. need still to switch execution order to get reports to github ui.

Reply:
Yeah, I mentioned that in my review too.

Reply:
this might help:

```sh
rc=0
command1 || rc=1
command2 || rc=2
exit $rc
```